Alexis docs hackathon #1 Aug 23 (#736)
CliMA/ClimaLand.jl: docs/src/getting-started.md

# Getting Started

## For Users

### Installation

First, download and install Julia by following the instructions at [https://julialang.org/downloads/](https://julialang.org/downloads/).
Then, you can install the ClimaLand package by doing:

```julia
julia> ] # Enter Package REPL mode
Pkg> add ClimaLand # Install ClimaLand
Pkg> # Go back to Julia REPL mode
Julia> using ClimaLand
```

> 

Reviewer: The transition from "Installation" to "Parametrisation" feels abrupt. 

Let's add a transition, like:
```
A typical land simulation employs several different parameterizations to model the various land-surface processes. Let's start our journet into ClimaLand by looking at one of those.
```

Author: Thank you!! I like that

> ### Parameterisation

Reviewer: I would spell it `Parameterization` (en-US instead en-GB)

> Let's start with an basic example: compute canopy gross photosynthesis (GPP).

Reviewer: ```suggestion
Let's start with a basic example: compute canopy gross photosynthesis (GPP).
```

```@repl
using ClimaLand
@doc ClimaLand.Canopy.compute_GPP
```

> As you can see, our parameterisation for GPP is located in the `Canopy` Module, and requires four FT.

Reviewer: ```suggestion
As you can see, our parameterisation for GPP is located in the [`Canopy`](@ref) Module, and requires four FT.
```

> As you can see, our parameterisation for GPP is located in the `Canopy` Module, and requires four FT.

Reviewer: I think users should not be worried about `FT` at this point. Let's introduce this complication later.

> As you can see, our parameterisation for GPP is located in the `Canopy` Module, and requires four FT.
> FT are Float (either Float32 or Float64). For example, with An = 5 µmol m⁻² s⁻¹, K = 0.5, LAI = 3 m² m⁻²,
> Ω = 0.7, you can compute GPP like below:

Reviewer: ```suggestion
As you can see, our parameterisation for GPP is located in the `Canopy` Module, and requires four FT.
FT means Float Type (typically Float32 or Float64). For example, with An = 5 µmol m⁻² s⁻¹, K = 0.5, LAI = 3 m² m⁻²,
Ω = 0.7, you can compute GPP like below:
```

```@repl
import ClimaLand.Canopy as canopy
FT = Float64
canopy.compute_GPP(FT(5.0), FT(0.5), FT(3.0), FT(0.7))
```

Et voilà!

> Note that our package [ParamViz](https://github.com/CliMA/ParamViz.jl) allow interactive visualisation of

Reviewer: ```suggestion
Note that our package [ParamViz](https://github.com/CliMA/ParamViz.jl) allows interactive visualisation of
```

        parts = split(full_name, ".")
        name = parts[end]  # Use only the last part of the name
        module_dict[full_name] = ModuleNode(name, [])
    end

    # Second pass: build the tree structure
    root = nothing
    for (full_name, node) in module_dict
        parts = split(full_name, ".")
        if length(parts) == 1
            root = node
        else
            parent_name = join(parts[1:end-1], ".")
            if haskey(module_dict, parent_name)
                push!(module_dict[parent_name].children, node)
            end
        end
    end

    return root
end

# Custom print function to sort children
function print_sorted_tree(io::IO, tree::ModuleNode, prefix::String="")
    println(io, prefix, nodevalue(tree))
    ch = sort(collect(children(tree)), by=x -> x.name)
    for (i, child) in enumerate(ch)
        if i == length(ch)
            print_sorted_tree(io, child, prefix * "└─ ")
        else
            print_sorted_tree(io, child, prefix * "├─ ")
        end
    end
end

# Get the modules
modules = child_modules(ClimaLand)

# Build the tree
root = build_tree(modules)
```

```@example climatree_md
print_sorted_tree(stdout, root) # hide
```

> To explore what modules, functions and types are exported in a particular module, you can use About.jl:

Reviewer: Please, add a link to `About.jl`.

```@repl
using ClimaLand
using About
about(ClimaLand.Soil.Biogeochemistry)
```

> Where names in red are module, names in blue are functions, and name in yellow are Types.

Reviewer: Let's not mention this until it is actually true.

# Getting Started

## For Users

### Installation

> Installing ClimaLand in Julia is very easy. First, download and install Julia

Reviewer: Describing something as "very easy" is typically not the best for pedagogy. It may be very easy for you, but it is certainly not very easy for my grandma. It might also lead to frustration: how would you feel if you couldn't accomplish a task described as "very easy"? I suggest we avoid language like this when possible to make our docs more welcoming.

Author: I agree! I adjusted the text
